tests/phpunit/Unit/Connection/CallbackConnectionProviderTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace SMW\Tests\Connection;
+
+use SMW\Connection\CallbackConnectionProvider;
+
+/**
+ * @covers \SMW\Connection\CallbackConnectionProvider
+ * @group semantic-mediawiki
+ *
+ * @license GNU GPL v2+
+ * @since 3.0
+ *
+ * @author mwjames
+ */
+class CallbackConnectionProviderTest extends \PHPUnit_Framework_TestCase {
+
+	public function testCanConstruct() {
+
+		$callback = function() {};
+
+		$this->assertInstanceOf(
+			CallbackConnectionProvider::class,
+			new CallbackConnectionProvider( $callback )
+		);
+	}
+
+	public function testGetConnection() {
+
+		$connection = $this->getMockBuilder( '\stdClass' )
+			->disableOriginalConstructor()
+			->getMock();
```

Comment:
huh?! This is the same as `$connection = new \stdClass()` no?

Reply:
Also beware that this PHPUnit syntax is not compatible with the latest versions of PHPUnit. I don't know if it is easy to switch to using a more recent version of PHPUnit (with SMW being a MW extension and all...), though if possible, doing this sooner rather than later will save migration work.

Reply:
This should work in any version, or? and if not what changed?

> to switch to using a more recent version of PHPUnit (with SMW being a MW extension and all...)

2713 has the details!

Reply:
https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/2713

Reply:
Oh... actually this might indeed *work* in the latest PHPUnit. I'm not sure as I have not used this format recently. Which is because it does not make sense to use this form there. You'd just do `$connection = $this->createMock( '\stdClass' )`

At least you would if there was reason to use the mock API. Why not just create the instance like in my first comment?